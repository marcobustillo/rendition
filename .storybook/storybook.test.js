import initStoryshots, { multiSnapshotWithOptions } from '@storybook/addon-storyshots'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
})

initStoryshots({
  storyKindRegex: /^Next\/Form$/,
  test: multiSnapshotWithOptions({
    createNodeMock: element => {
      if (element.type === 'input' && element.props.type === 'file') {
        return document.createElement('input')
      }
    }
  })
})
