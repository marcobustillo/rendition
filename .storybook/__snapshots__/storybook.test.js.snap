// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Core/Accordion Default 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c14 {
  -webkit-transition: max-height 200ms,opacity 200ms;
  transition: max-height 200ms,opacity 200ms;
  opacity: 0;
  overflow: hidden;
  max-height: 0;
}

.c0 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c5 {
  box-sizing: border-box;
}

.c7 {
  padding-left: 16px;
  padding-right: 16px;
}

.c11 {
  padding-right: 16px;
}

.c15 {
  margin-top: 10px;
  margin-bottom: 24px;
  padding-left: 16px;
  padding-right: 16px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c9 {
  font-weight: 600;
}

.c10 {
  margin-top: 24px;
  margin-bottom: 24px;
  font-size: 16px;
}

.c1 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

@media only screen and (max-width:768px) {
  .c13 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

<div
  className="c0 c1"
>
  <div
    className="c2"
    role="tablist"
  >
    <div
      className="c3"
    >
      <button
        aria-expanded={false}
        aria-selected={false}
        className="c4"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        role="tab"
        type="button"
      >
        <div
          className="c5 c6"
        >
          <div
            className="c5 c7 c8"
          >
            <div
              className="c9 c10"
            >
              Heading one
            </div>
          </div>
          <div
            className="c5 c11 c12"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 "
              color="#c6c8c9"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </button>
      <div
        className="c13"
      >
        <div
          aria-hidden={true}
          className="c2 c14"
          open={false}
        >
          <div
            className="c5 c15"
          >
            First Panel
          </div>
        </div>
      </div>
    </div>
    <div
      className="c3"
    >
      <button
        aria-expanded={false}
        aria-selected={false}
        className="c4"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        role="tab"
        type="button"
      >
        <div
          className="c5 c6"
        >
          <div
            className="c5 c7 c8"
          >
            <div
              className="c9 c10"
            >
              Second Heading
            </div>
          </div>
          <div
            className="c5 c11 c12"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 "
              color="#c6c8c9"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </button>
      <div
        className="c13"
      >
        <div
          aria-hidden={true}
          className="c2 c14"
          open={false}
        >
          <div
            className="c5 c15"
          >
            Second Panel
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Core/Accordion Primary 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c14 {
  -webkit-transition: max-height 200ms,opacity 200ms;
  transition: max-height 200ms,opacity 200ms;
  opacity: 0;
  overflow: hidden;
  max-height: 0;
}

.c0 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c5 {
  box-sizing: border-box;
}

.c7 {
  padding-left: 16px;
  padding-right: 16px;
}

.c11 {
  padding-right: 16px;
}

.c15 {
  margin-top: 10px;
  margin-bottom: 24px;
  padding-left: 16px;
  padding-right: 16px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c9 {
  font-weight: 600;
}

.c10 {
  margin-top: 24px;
  margin-bottom: 24px;
  font-size: 16px;
}

.c1 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

@media only screen and (max-width:768px) {
  .c13 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

<div
  className="c0 c1"
>
  <div
    className="c2"
    role="tablist"
  >
    <div
      className="c3"
    >
      <button
        aria-expanded={false}
        aria-selected={false}
        className="c4"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        role="tab"
        type="button"
      >
        <div
          className="c5 c6"
        >
          <div
            className="c5 c7 c8"
          >
            <div
              className="c9 c10"
            >
              Heading one
            </div>
          </div>
          <div
            className="c5 c11 c12"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 "
              color="#c6c8c9"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </button>
      <div
        className="c13"
      >
        <div
          aria-hidden={true}
          className="c2 c14"
          open={false}
        >
          <div
            className="c5 c15"
          >
            First Panel
          </div>
        </div>
      </div>
    </div>
    <div
      className="c3"
    >
      <button
        aria-expanded={false}
        aria-selected={false}
        className="c4"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        role="tab"
        type="button"
      >
        <div
          className="c5 c6"
        >
          <div
            className="c5 c7 c8"
          >
            <div
              className="c9 c10"
            >
              Second Heading
            </div>
          </div>
          <div
            className="c5 c11 c12"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 "
              color="#c6c8c9"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </button>
      <div
        className="c13"
      >
        <div
          aria-hidden={true}
          className="c2 c14"
          open={false}
        >
          <div
            className="c5 c15"
          >
            Second Panel
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Core/Alert Default 1`] = `
.c0 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  color: #1AC135;
}

.c7 {
  color: #138b27;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin: 0;
  padding: 16px;
  padding-right: 16px;
  box-shadow: 0 0 10px 0 rgba(221,225,240,0.15);
  border-radius: 10px;
  border: 1px solid #DDE1f0;
  background-color: white;
  color: #1AC135;
}

.c8 {
  font-weight: 600;
}

.c5 {
  margin-bottom: auto;
  margin-right: 8px;
  color: #1AC135;
}

.c9 {
  margin-right: 4px;
}

.c10 {
  color: #2A506F;
}

.c1 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3 c4 "
  >
    <div
      className="c5"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-check-circle fa-w-16 "
        data-icon="check-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="c6 c7"
    >
      <span
        className="c8 c9"
      >
        Success!
      </span>
      <span
        className="c10"
      >
        This is a default alert
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Core/Alert Emphasized 1`] = `
.c0 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  color: #1496E1;
}

.c7 {
  color: #107dbc;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin: 0;
  padding: 16px;
  padding-right: 16px;
  box-shadow: 0 0 10px 0 rgba(221,225,240,0.15);
  border-radius: 10px;
  border: 1px solid #1496E1;
  background-color: #e8f5fc;
  color: #107dbc;
}

.c8 {
  font-weight: 600;
}

.c5 {
  margin-bottom: auto;
  margin-right: 8px;
  color: #1496E1;
}

.c9 {
  margin-right: 4px;
}

.c10 {
  color: #107dbc;
}

.c1 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3 c4 "
  >
    <div
      className="c5"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-lightbulb fa-w-11 "
        data-icon="lightbulb"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96.06 454.35c.01 6.29 1.87 12.45 5.36 17.69l17.09 25.69a31.99 31.99 0 0 0 26.64 14.28h61.71a31.99 31.99 0 0 0 26.64-14.28l17.09-25.69a31.989 31.989 0 0 0 5.36-17.69l.04-38.35H96.01l.05 38.35zM0 176c0 44.37 16.45 84.85 43.56 115.78 16.52 18.85 42.36 58.23 52.21 91.45.04.26.07.52.11.78h160.24c.04-.26.07-.51.11-.78 9.85-33.22 35.69-72.6 52.21-91.45C335.55 260.85 352 220.37 352 176 352 78.61 272.91-.3 175.45 0 73.44.31 0 82.97 0 176zm176-80c-44.11 0-80 35.89-80 80 0 8.84-7.16 16-16 16s-16-7.16-16-16c0-61.76 50.24-112 112-112 8.84 0 16 7.16 16 16s-7.16 16-16 16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="c6 c7"
    >
      <span
        className="c8 c9"
      >
        Note!
      </span>
      <span
        className="c10"
      >
        This is an emphasized alert
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Core/Alert Plaintext 1`] = `
.c0 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  box-sizing: border-box;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  margin-bottom: auto;
  margin-right: 8px;
  color: #1496E1;
}

.c5 {
  color: #107dbc;
}

.c1 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3 "
  >
    <div
      className="c4"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-lightbulb fa-w-11 "
        data-icon="lightbulb"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96.06 454.35c.01 6.29 1.87 12.45 5.36 17.69l17.09 25.69a31.99 31.99 0 0 0 26.64 14.28h61.71a31.99 31.99 0 0 0 26.64-14.28l17.09-25.69a31.989 31.989 0 0 0 5.36-17.69l.04-38.35H96.01l.05 38.35zM0 176c0 44.37 16.45 84.85 43.56 115.78 16.52 18.85 42.36 58.23 52.21 91.45.04.26.07.52.11.78h160.24c.04-.26.07-.51.11-.78 9.85-33.22 35.69-72.6 52.21-91.45C335.55 260.85 352 220.37 352 176 352 78.61 272.91-.3 175.45 0 73.44.31 0 82.97 0 176zm176-80c-44.11 0-80 35.89-80 80 0 8.84-7.16 16-16 16s-16-7.16-16-16c0-61.76 50.24-112 112-112 8.84 0 16 7.16 16 16s-7.16 16-16 16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <span
      className="c5"
    >
      This is a plaintext alert
    </span>
  </div>
</div>
`;
